### ISSUE TITLE
Publish dall-e-shell to MELPA

### ISSUE THREAD

**Issue by xenodium** [xenodium/chatgpt-shell#68]
Similar to https://github.com/melpa/melpa/pull/8479

- Update dall-e-shell.el to

Package-Requires: ((emacs "27.1") (shell-maker "0.17.1"))

- Send PR to chatgpt-shell repo. 

- Needs a fork of https://github.com/melpa/melpa

- Create melpa/recipes/dall-e-shell with:

```
(dall-e-shell
  :fetcher github
  :repo "xenodium/chatgpt-shell"
  :files ("dall-e-shell.el"))
```
- Ensure all MELPA checks pass (see chatgpt-shell PR with all initial checkboxes). 

- Send PR to melpa repo.

- Wait for approval or further feedback.

**Comment by NatHarari** [comment]
Thanks for the reply. I'm confused: I have to still download the dall-e shell separately, yes?

**Comment by xenodium** [comment]
As a user ChatGPT and DALL-E will need to be installed separately, so they need separate MELPA recipes. Does this answer the question or do we mean something else?

**Comment by NatHarari** [comment]
Well I installed your chatgpt-shell from melpa. I just need to install dall-e shell by downloading it separately, yes? Sorry for the confusion. I'm not a coder, and I don't play one on the internet. 😀

**Comment by xenodium** [comment]
ps. This separation is as per MELPA maintainer guidance. Otherwise all of the packages would need to share the same prefix (ie. chatgpt-shell) which would be weird for for dall-e-shell. Hope that makes more sense.

**Comment by NatHarari** [comment]
Yeah I'm a bit lost. I apologise. I did download dall-e-shell.el and I put it in ~/.emacs.d/packages/

Then I added the lines above to my config:
  (dall-e-shell
    :fetcher github
    :repo "xenodium/chatgpt-shell"
    :files ("~/.emacs.d/packages/dall-e-shell.el"))

Then I evaluated it and got an error. I may not be the one you're looking for, unfortunately.

**Comment by xenodium** [comment]
Installing from MELPA will only be possible after the MELPA pull request is accepted by MELPA maintainers. 

There are changes needed in two git repositories chatgpt-shell and MELPA. 

If you're keen to take the DALL-E MELPA submission, I can handle one of the other pending packages like ob-chatgpt-shell.el). I can include you in all the changes. Same changes would apply for DALL-E.

No pressure here. Feel free to take it or wait until I get to it ;)

**Comment by xenodium** [comment]
Ah no worries. This issue/item is tracking the submission to MELPA so it can be installed through it.

If you'd like to install outside of MELPA, download dall-e-shell.el to a location in your load path and require it.

**Comment by NatHarari** [comment]
Yeah that's what I did. it works beautifully. I'll stay out of your way for this but I do very much appreciate it. Thank you.

**Comment by xenodium** [comment]
Opened https://github.com/melpa/melpa/pull/8513
